src/components/SingleEvent/EventParticipants.js

```diff
@@ -47,13 +56,41 @@ class EventParticipants extends Component {
       <Fragment>
         <H3>Participants - <Spots>{spots}</Spots></H3>
         <EventFilters handleSearch={handleSearch} />
+        {amAdmin? (
+          <SafeQuery
+                query={QRSupportedQuery}
+                variables={{ address: '1' }}
+              >
+                {result => {
+                  if (result.data.scanQRCodeSupported && result.data.scanQRCodeSupported.supported) {
+                    return(
+                      <SafeMutation mutation={ScanQRCode}>
+                        {scanQRCode => (
+                          <QRCodeContainer>
+                            <Button 
+                              onClick={ (() => {
```

Comment:
```suggestion
                              onClick={this._scan({ scanQRCode, setSearchTerm })
```

Then in `EventParticipants` class:

```
_scan = ({ scanQRCode, setSearchTerm }) => async () => {
  const result =await scanQRCode()
  const code = _.get(result, 'data.scanQRCode.address')
  if (code) {
    setSearchTerm(code)
  }
}
```

Note that `_` = `lodash` module